In [1]:
from IPython.display import display, HTML
display(HTML("<style>.container { width:90% !important; }</style>"))

# 작업 1유형

### 웹 광고 눌러주세요 ㅎㅎ

- 주피터 노트북 사용법에 대한 설명 x
- `작업 1유형 연습문제` 114문제 풀이
- 문법에 대한 깊이 있는 설명은 `전처리 100문제` 참고

### 유튜브 인기동영상 데이터 

In [2]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/youtube/youtube.csv",index_col=0)
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 60399 entries, 0 to 60398
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   title           60399 non-null  object
 1   channelTitle    60399 non-null  object
 2   categoryId      60399 non-null  int64 
 3   view_count      60399 non-null  int64 
 4   likes           60399 non-null  int64 
 5   dislikes        60399 non-null  int64 
 6   comment_count   60399 non-null  int64 
 7   channelId       60399 non-null  object
 8   trending_date2  60399 non-null  object
dtypes: int64(5), object(4)
memory usage: 4.6+ MB


## Question 1
### **인기동영상 제작횟수가 많은 채널 상위 10개명을 출력하라 (날짜기준, 중복포함)**

In [3]:
top10 = df.loc[df['channelId'].isin(df.groupby('channelId').size().reset_index(name='counts').sort_values(by='counts', ascending=False).head(10)['channelId'])]['channelTitle'].unique()
print(top10)

['장삐쭈' '총몇명' '파뿌리' '짤툰' '런닝맨 - 스브스 공식 채널' '엠뚜루마뚜루 : MBC 공식 종합 채널' 'SPOTV'
 '채널 십오야' '이과장' 'BANGTANTV']


In [4]:
answer =list(df.loc[df.channelId.isin(df.channelId.value_counts().head(10).index)].channelTitle.unique())
print(answer)

['장삐쭈', '총몇명', '파뿌리', '짤툰', '런닝맨 - 스브스 공식 채널', '엠뚜루마뚜루 : MBC 공식 종합 채널', 'SPOTV', '채널 십오야', '이과장', 'BANGTANTV']


## Question 2
### **논란으로 인기동영상이 된 케이스를 확인하고 싶다. dislikes수가 like 수보다 높은 동영상을 제작한 채널을 모두 출력하라**

In [5]:
df.columns

Index(['title', 'channelTitle', 'categoryId', 'view_count', 'likes',
       'dislikes', 'comment_count', 'channelId', 'trending_date2'],
      dtype='object')

In [6]:
answer = list(df.loc[df.likes < df.dislikes].channelTitle.unique())
print(answer)

['핫도그TV', 'ASMR 애정TV', '하얀트리HayanTree', '양팡 YangPang', '철구형 (CHULTUBE)', '왜냐맨하우스', '(MUTUBE)와꾸대장봉준', '오메킴TV', '육지담', 'MapleStory_KR', 'ROAD FIGHTING CHAMPIONSHIP', '사나이 김기훈', '나혼자산다 STUDIO', 'Gen.G esports']


## Question 3
### **채널명을 바꾼 케이스가 있는지 확인하고 싶다. channelId의 경우 고유값이므로 이를 통해 채널명을 한번이라도 바꾼 채널의 갯수를 구하여라**

In [7]:
change = df[['channelId','channelTitle']].drop_duplicates().channelId.value_counts()
target = change[change>1]
print(len(target))

71


## Question 4
### **일요일에  인기있었던 영상들중 가장많은 영상 종류(categoryId)는 무엇인가?**

In [8]:
df['trending_date2'] = df['trending_date2'].astype('datetime64')

In [9]:
df.loc[df.trending_date2.dt.day_name() == 'Sunday'].categoryId.value_counts().index[0]

24

## Question 5
### **각 요일별 인기 영상들의 categoryId는 각각 몇개 씩인지 하나의 데이터 프레임으로 표현하라**

In [10]:
Ans = df.pivot_table ( index = 'categoryId' , columns = df.trending_date2.dt.day_name() , aggfunc = 'size')
Ans

trending_date2  Friday  Monday  Saturday  Sunday  Thursday  Tuesday  Wednesday
categoryId                                                                    
1                  243     263       255     274       246      257        234
2                  120     105       119      99       128      119        129
10                 833     837       776     830       890      894        917
15                 187     215       198     217       207      208        207
17                 633     668       592     636       682      708        706
19                  90      92        87      91        92       89         85
20                 283     298       296     289       282      285        291
22                1288    1373      1289    1337      1341     1375       1333
23                 568     594       570     556       560      569        566
24                2976    3148      3066    3096      2954     3084       3090
25                 444     453       422     437       470      452        468
26                 369     378       364     363       375      394        385
27                 183     205       183     199       194      194        212
28                 171     160       173     167       166      161        165
29                  12      10        10       9        13       11         12

## Question 6
### **댓글의 수로 (comment_count) 영상 반응에 대한 판단을 할 수 있다. viewcount대비 댓글수가 가장 높은 영상을 확인하라 (view_count값이 0인 경우는 제외한다)**

In [11]:
target2 = df.loc[df.view_count != 0]
t = target2.copy()
t['ratio'] = (target2['comment_count']/target2['view_count']).dropna()
result = t.sort_values(by = 'ratio', ascending = False).iloc[0].title
print(result)

60분 동안 댓글이 달리지 않으면, 영상이 삭제됩니다. (챌린지)


## Question 7
### **댓글의 수로 (comment_count) 영상 반응에 대한 판단을 할 수 있다.viewcount대비 댓글수가 가장 낮은 영상을 확인하라 (view_counts, ratio값이 0인경우는 제외한다.)**

In [12]:
result = t.sort_values(by = 'ratio' , ascending = True)
result = result.loc[ result.ratio != 0].iloc[0].title
print(result)

Join the BTS #PermissiontoDance Challenge only on YouTube #Shorts


## Question 8
### **like 대비 dislike의 수가 가장 적은 영상은 무엇인가? (like, dislike 값이 0인경우는 제외한다)**

In [13]:
target1 = df.loc[(df.likes != 0) & (df.dislikes != 0)]
t2 = target1.copy()
t2['ratio'] = (target1['dislikes']/target1['likes']).dropna()
t2.sort_values(by = 'ratio').iloc[0].title

'[줌터뷰] *최초공개* 사부작즈🐰🐶의 비공식 이름은 아이라인즈? 꿀조합 티키타카 가득한 NCT 127 도영&정우의 줌터뷰'

## Question 9
### **가장많은 트렌드 영상을 제작한 채널의 이름은 무엇인가? (날짜기준, 중복포함)**

In [14]:
Ans = df.loc[df.channelId == df.channelId.value_counts().index[0]].channelTitle.unique()
print(Ans)

['짤툰']


## Question 10
### **20회(20일)이상 인기동영상 리스트에 포함된 동영상의 숫자는?**

In [15]:
answer = (df[['title','channelId']].value_counts() >= 20).sum()
print(answer)

40


### 유튜브 공범컨텐츠 동영상 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/kukuroo3/youtube-episodic-contents-kr(참고, 데이터 수정)     
데이터 설명 : 유튜브 "공범" 컨텐츠 동영상 정보 ( 10분 간격 수집)    
dataurl1 (비디오 정보) =    https://raw.githubusercontent.com/Datamanim/datarepo/main/youtube/videoInfo.csv    
dataurl2 (참가자 채널 정보)=   https://raw.githubusercontent.com/Datamanim/datarepo/main/youtube/channelInfo.csv      
```

In [16]:
import pandas as pd


channel =pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/youtube/channelInfo.csv')
video =pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/youtube/videoInfo.csv')
display(channel.head())
display(video.head())



channelid   subcnt    viewcnt  videocnt  \
0  UCkQCwnkQfgSuPTTnw_Y7v7w  1310000  410238653       736   
1  UCkQCwnkQfgSuPTTnw_Y7v7w  1310000  412531322       736   
2  UCkQCwnkQfgSuPTTnw_Y7v7w  1310000  412531322       735   
3  UCkQCwnkQfgSuPTTnw_Y7v7w  1310000  412531322       737   
4  UCkQCwnkQfgSuPTTnw_Y7v7w  1320000  412531322       737   

                    ct channelname  
0  2021-09-30 03:01:03         꽈뚜룹  
1  2021-09-30 09:01:03         꽈뚜룹  
2  2021-09-30 15:01:03         꽈뚜룹  
3  2021-09-30 21:01:03         꽈뚜룹  
4  2021-10-01 03:01:04         꽈뚜룹

videopk  viewcnt  likecnt  dislikecnt  favoritecnt  cmcnt  \
0  c5JQp6xafqc  1667010    30474         706            0   6587   
1  c5JQp6xafqc  1669089    30495         707            0   6589   
2  c5JQp6xafqc  1674759    30522         711            0   6596   
3  c5JQp6xafqc  1677026    30555         712            0   6604   
4  c5JQp6xafqc  1681824    30585         713            0   6600   

                    ct videoname  
0  2021-10-10 15:20:03    공범 EP1  
1  2021-10-10 15:30:03    공범 EP1  
2  2021-10-10 15:40:02    공범 EP1  
3  2021-10-10 15:50:03    공범 EP1  
4  2021-10-10 16:00:03    공범 EP1

## Question 11
### **각 데이터의 'ct'컬럼을 시간으로 인식할수 있게 datatype을 변경하고 video 데이터의 videoname의 각 value 마다 몇개의 데이터씩 가지고 있는지 확인하라**

In [17]:
channel['ct'] = channel.ct.astype('datetime64')
video['ct'] = video.ct.astype('datetime64')

In [18]:
video.videoname.value_counts()

 공범 EP1    3492
 공범 EP2    3204
 공범 EP3    2568
 공범 EP4    2280
 공범 EP5    1562
 공범 EP6    1274
 공범 EP7     555
 공범 EP8     266
Name: videoname, dtype: int64

## Question 12
### **수집된 각 video의 가장 최신화 된 날짜의 viewcount값을 출력하라**

In [19]:
video.sort_values(['videoname','ct']).drop_duplicates('videoname', keep = 'last')[['viewcnt','videoname','ct']].reset_index(drop=True)

viewcnt videoname                  ct
0  3180532    공범 EP1 2021-11-01 15:30:03
1  2199328    공범 EP2 2021-11-01 15:30:03
2  1671294    공범 EP3 2021-11-01 15:30:03
3  1818493    공범 EP4 2021-11-01 15:30:03
4  1503435    공범 EP5 2021-11-01 15:30:04
5  1750222    공범 EP6 2021-11-01 15:30:04
6  1630200    공범 EP7 2021-11-01 15:30:05
7  1289088    공범 EP8 2021-11-01 15:30:05

## Question 13
### **Channel 데이터중 2021-10-03일 이후 각 채널의 처음 기록 됐던 구독자 수(subcnt)를 출력하라**

In [20]:
channel.loc[channel.ct >= pd.to_datetime('2021-10-03')].sort_values(['ct', 'channelname']).drop_duplicates('channelname', keep = 'first')[['channelname','subcnt']].reset_index(drop = True)

channelname   subcnt
0            논리왕 전기   922000
1             야전삽짱재   257000
2        츄정ChuJeong   322000
3        김농밀의 농밀한 삶     7520
4       릴펄 Lilpearl    10100
5   와글와글 WagleWagle    55000
6               조나단   215000
7           형사!탐정되다    14900
8     Balming Tiger    54300
9     곽토리 kwak tori   471000
10              꽈뚜룹  1330000

## Question 14
### **각채널의 2021-10-03 03:00:00 ~ 2021-11-01 15:00:00 까지 구독자수 (subcnt) 의 증가량을 구하여라**

In [21]:
start = channel.loc[channel.ct >= pd.to_datetime('2021-10-03 03')].sort_values(['ct','channelname']).drop_duplicates('channelname', keep = 'first')[['subcnt','channelname']].reset_index(drop=True)
end = channel.loc[channel.ct <= pd.to_datetime('2021-11-01 15')].sort_values(['ct','channelname'],ascending = False).drop_duplicates('channelname', keep = 'first')[['subcnt','channelname']].reset_index(drop=True)
start.columns = ['start_sub','channelname']
end.columns = ['end_sub','channelname']
answer = pd.merge(start,end)
answer['del'] = answer['end_sub'] - answer['start_sub']
print(answer[['channelname','del']])

        channelname    del
0            논리왕 전기 -11000
1             야전삽짱재  11000
2        츄정ChuJeong   1000
3        김농밀의 농밀한 삶   1540
4       릴펄 Lilpearl  11000
5   와글와글 WagleWagle      0
6               조나단  12000
7           형사!탐정되다  10300
8     Balming Tiger   2500
9     곽토리 kwak tori  -2000
10              꽈뚜룹  70000


## Question 15
### **각 비디오는 10분 간격으로 구독자수, 좋아요, 싫어요수, 댓글수가 수집된것으로 알려졌다. 공범 EP1의 비디오정보 데이터중 수집간격이 5분 이하, 20분이상인 데이터 구간( 해당 시점 전,후) 의 시각을 모두 출력하라**

In [22]:
video.ct.diff()

0                   NaT
1       0 days 00:10:00
2       0 days 00:09:59
3       0 days 00:10:01
4       0 days 00:10:00
              ...      
15196   0 days 00:09:58
15197   0 days 00:09:59
15198   0 days 00:10:00
15199   0 days 00:09:59
15200   0 days 00:10:01
Name: ct, Length: 15201, dtype: timedelta64[ns]

In [23]:
video.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15201 entries, 0 to 15200
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   videopk      15201 non-null  object        
 1   viewcnt      15201 non-null  int64         
 2   likecnt      15201 non-null  int64         
 3   dislikecnt   15201 non-null  int64         
 4   favoritecnt  15201 non-null  int64         
 5   cmcnt        15201 non-null  int64         
 6   ct           15201 non-null  datetime64[ns]
 7   videoname    15201 non-null  object        
dtypes: datetime64[ns](1), int64(5), object(2)
memory usage: 950.2+ KB


In [24]:
video.sort_values('ct')

videopk  viewcnt  likecnt  dislikecnt  favoritecnt  cmcnt  \
10     c5JQp6xafqc     6396      680           3            0    226   
11     c5JQp6xafqc    13087     1113           7            0    273   
12     c5JQp6xafqc    19991     1446          12            0    363   
13     c5JQp6xafqc    27451     1859          19            0    479   
14     c5JQp6xafqc    36559     2306          23            0    602   
...            ...      ...      ...         ...          ...    ...   
6695   23QQ1ru9YQg  2199328    25566         763            0   9171   
13105  hv7FBjskAu0  1503435    19926         421            0  12434   
14934  AxGKHmXyuAE  1750222    24494         389            0  13025   
13660  QUjQbgj5IXM  1630200    29335         400            0  18247   
15200  yZt-h-KcmUE  1289088    25616        3621            0  31663   

                       ct videoname  
10    2021-10-07 19:10:03    공범 EP1  
11    2021-10-07 19:20:02    공범 EP1  
12    2021-10-07 19:30:03    공범 EP1  
13    2021-10-07 19:40:03    공범 EP1  
14    2021-10-07 19:50:03    공범 EP1  
...                   ...       ...  
6695  2021-11-01 15:30:03    공범 EP2  
13105 2021-11-01 15:30:04    공범 EP5  
14934 2021-11-01 15:30:04    공범 EP6  
13660 2021-11-01 15:30:05    공범 EP7  
15200 2021-11-01 15:30:05    공범 EP8  

[15201 rows x 8 columns]

In [25]:
import datetime
video1 = video.loc[video.videoname == ' 공범 EP1'].sort_values('ct')
ep1 = video1.loc[(video1.ct.diff(1) >= datetime.timedelta(minutes = 20)) | (video1.ct.diff(1) <= datetime.timedelta(minutes = 5))]
video1.loc[video1.index.isin(ep1.index-1) | video1.index.isin(ep1.index) | video1.index.isin(ep1.index+1)]

videopk  viewcnt  likecnt  dislikecnt  favoritecnt  cmcnt  \
720   c5JQp6xafqc  2153676    34104         830            0   7120   
721   c5JQp6xafqc  2228250    34559         849            0   7191   
722   c5JQp6xafqc  2228250    34559         849            0   7191   
723   c5JQp6xafqc  2228669    34566         849            0   7194   
1635  c5JQp6xafqc  2706691    37893         987            0   7410   
1636  c5JQp6xafqc  2707933    37901         988            0   7410   
1637  c5JQp6xafqc  2708448    37903         988            0   7410   

                      ct videoname  
720  2021-10-12 19:20:03    공범 EP1  
721  2021-10-13 09:41:37    공범 EP1  
722  2021-10-13 09:41:37    공범 EP1  
723  2021-10-13 09:50:03    공범 EP1  
1635 2021-10-19 17:50:02    공범 EP1  
1636 2021-10-19 18:20:03    공범 EP1  
1637 2021-10-19 18:30:03    공범 EP1

## Question 16
### **각 에피소드의 시작날짜(년-월-일)를 에피소드 이름과 묶어 데이터 프레임으로 만들고 출력하라**

In [26]:
v1 = video.copy()
v1['date'] = v1.ct.dt.date
v1.sort_values('ct').drop_duplicates('videoname', keep = 'first')[['date','videoname']]

date videoname
10     2021-10-07    공범 EP1
3496   2021-10-09    공범 EP2
6696   2021-10-14    공범 EP3
9264   2021-10-16    공범 EP4
11544  2021-10-21    공범 EP5
13661  2021-10-23    공범 EP6
13106  2021-10-28    공범 EP7
14935  2021-10-30    공범 EP8

## Question 17
### **"공범" 컨텐츠의 경우 19:00시에 공개 되는것으로 알려져있다. 공개된 날의 21시의 viewcnt, ct, videoname 으로 구성된 데이터 프레임을 viewcnt를 내림차순으로 정렬하여 출력하라**

In [27]:
import time

In [28]:
%%timeit
video['time']= video.ct.dt.hour

answer = video.loc[video['time'] ==21] \
            .sort_values(['videoname','ct'])\
            .drop_duplicates('videoname') \
            .sort_values('viewcnt',ascending=False)[['videoname','viewcnt','ct']]\
            .reset_index(drop=True)

5.88 ms ± 460 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


In [29]:
%%timeit
video.loc[video.ct.dt.hour == 21].sort_values('ct').drop_duplicates('videoname', keep = 'first')[['videoname', 'viewcnt', 'ct']].sort_values('viewcnt', ascending = False).reset_index(drop=True)

3.77 ms ± 114 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


## Question 18
### **video 정보의 가장 최근 데이터들에서 각 에피소드의 싫어요/좋아요 비율을 ratio 컬럼으로 만들고 videoname, ratio로 구성된 데이터 프레임을 ratio를 오름차순으로 정렬하라**

In [30]:
v1 = video.sort_values(['ct','videoname'], ascending = False).drop_duplicates('videoname', keep = 'first')
v1['ratio'] = (v1.dislikecnt) / (v1.likecnt)
v1[['videoname','ratio']].sort_values('ratio').reset_index(drop = True)

videoname     ratio
0    공범 EP7  0.013636
1    공범 EP6  0.015881
2    공범 EP4  0.017971
3    공범 EP5  0.021128
4    공범 EP3  0.022360
5    공범 EP1  0.027118
6    공범 EP2  0.029844
7    공범 EP8  0.141357

## Question 19
### **2021-11-01 00:00:00 ~ 15:00:00까지 각 에피소드별 viewcnt의 증가량을 데이터 프레임으로 만드시오**

In [31]:
target = video.loc[(video.ct >= pd.to_datetime('2021-11-01 00:00:00')) & (video.ct <= pd.to_datetime('2021-11-01 15:00:00'))].sort_values(['videoname','ct']).reset_index(drop = True)

def check(x) :
    result = max(x) - min(x)
    return result

target[['videoname','viewcnt']].groupby('videoname').agg(check)

viewcnt
videoname         
 공범 EP1      13298
 공범 EP2      10300
 공범 EP3       9927
 공범 EP4       9824
 공범 EP5      10824
 공범 EP6      14141
 공범 EP7      26949
 공범 EP8      89147

## Question 20
### **video 데이터 중에서 중복되는 데이터가 존재한다. 중복되는 각 데이터의 시간대와  videoname 을 구하여라**

In [41]:
answer = video[video.index.isin(set(video.index) - set(video.drop_duplicates().index))]
result = answer[['videoname','ct']]
display(result)

videoname                  ct
722     공범 EP1 2021-10-13 09:41:37
3927    공범 EP2 2021-10-13 09:41:37

In [39]:
video.loc[ ~video.index.isin(video.drop_duplicates().index)][['videoname','ct']]

videoname                  ct
722     공범 EP1 2021-10-13 09:41:37
3927    공범 EP2 2021-10-13 09:41:37

### 월드컵 출전선수 골기록 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/darinhawley/fifa-world-cup-goalscorers-19302018(참고, 데이터 수정)     
데이터 설명 : 1930 ~2018년도 월드컵 출전선수 골기록     
data url = https://raw.githubusercontent.com/Datamanim/datarepo/main/worldcup/worldcupgoals.csv      
```

In [40]:
import pandas as pd

df= pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/worldcup/worldcupgoals.csv')
df.head()

Player  Goals                Years  Country
0  Miroslav Klose     16  2002-2006-2010-2014  Germany
1         Ronaldo     15       1998-2002-2006   Brazil
2     Gerd Muller     14            1970-1974  Germany
3   Just Fontaine     13                 1958   France
4            Pele     12  1958-1962-1966-1970   Brazil

## Question 21
### **주어진 전체 기간의 각 나라별 골득점수 상위 5개 국가와 그 득점수를 데이터프레임형태로 출력하라**

In [51]:
df.groupby('Country').sum('Goals').sort_values('Goals', ascending = False).head(5)

Goals
Country         
Brazil       228
Germany      226
Argentina    135
Italy        123
France       115

## Question 22
### **주어진 전체기간동안 골득점을 한 선수가 가장 많은 나라 상위 5개 국가와 그 선수 숫자를 데이터 프레임 형식으로 출력하라**

In [63]:
df.groupby('Country').size().sort_values(ascending = False).head(5)

Country
Brazil       81
Germany      78
Italy        60
Argentina    59
France       58
dtype: int64

## Question 23
### **Years 컬럼은 년도-년도 형식으로 구성되어있고, 각 년도는 4자리 숫자이다. 년도 표기가 4자리 숫자로 안된 케이스가 존재한다.**
### **해당 건은 몇 건인지 출력하라**

In [90]:
df.Years.str.split('-')

0       [2002, 2006, 2010, 2014]
1             [1998, 2002, 2006]
2                   [1970, 1974]
3                         [1958]
4       [1958, 1962, 1966, 1970]
                  ...           
1290                      [1962]
1291                [1982, 1990]
1292                      [1930]
1293                      [1930]
1294                [1954, 1958]
Name: Years, Length: 1295, dtype: object

In [89]:
df['yearLst'] = df.Years.str.split('-')

def checkFour(x) :
    for value in x :
        if len(str(value)) !=4 :
            return False
    return True

df['check'] = df['yearLst'].apply(checkFour)
len(df[df.check == False])

45

## Question 24
### Q3에서 발생한 예외 케이스를 제외한 데이터프레임을  df2라고 정의하고 데이터의 행의 숫자를 출력하라 
(아래 문제부터는 df2로 풀이하겠습니다) 

In [112]:
df2 = df.loc[df.check == True ]
df2.shape[0]

1250

## Question 25
### **월드컵 출전횟수를 나타내는 'LenCup' 컬럼을 추가하고 4회 출전한 선수의 숫자를 구하여라**

In [151]:
df2.loc[:,'LenCup'] = df2['Years'].str.split('-').str.len()
df2.LenCup.value_counts()[4]

C:\Users\USER-PC\AppData\Local\Temp\ipykernel_31640\1286042677.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2.loc[:,'LenCup'] = df2['Years'].str.split('-').str.len()


16

## Question 26
### **Yugoslavia 국가의 월드컵 출전횟수가 2회인 선수들의 숫자를 구하여라**

In [160]:
len(df2.loc[(df2.Country == 'Yugoslavia') & (df2.LenCup == 2)])

7

## Question 27
### **2002년도에 출전한 전체 선수는 몇명인가?**

In [179]:
len(df2[df2.Years.str.contains('2002')])

156

## Question 28
### **이름에 'carlos' 단어가 들어가는 선수의 숫자는 몇 명인가? (대, 소문자 구분 x)**

In [184]:
len(df2[df2.Player.str.lower().str.contains('carlos')])

13

## Question 29
### **월드컵 출전 횟수가 1회뿐인 선수들 중에서 가장 많은 득점을 올렸던 선수는 누구인가?**

In [195]:
df2.loc[df2.LenCup == 1].sort_values('Goals', ascending = False).Player.values[0]

'Just Fontaine'

## Question 30
### **월드컵 출전횟수가 1회 뿐인 선수들이 가장 많은 국가는 어디인가?**

In [212]:
df2.loc[df2.LenCup == 1].groupby('Country').size().sort_values(ascending =False).head(1).index[0]

'Brazil'

### 서울시 따릉이 이용정보 데이터

```{attention} 
데이터 출처 :https://www.data.go.kr/data/15051872/fileData.do(참고, 데이터 수정)     
데이터 설명 : 서울특별시_공공자전거 시간대별 이용정보    
data url = https://raw.githubusercontent.com/Datamanim/datarepo/main/bicycle/seoul_bi.csv     
```

In [215]:
import pandas as pd
df =pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/bicycle/seoul_bi.csv')
df.head()

대여일자  대여시간  대여소번호 대여구분코드 성별 연령대코드  이용건수    운동량   탄소량     이동거리  사용시간
0  2021-06-01     0   3541    정기권  F  ~10대     1   0.00  0.00     0.00     8
1  2021-06-01     0    765    정기권  F  ~10대     1  27.21  0.35  1526.81    19
2  2021-06-01     0   2637    정기권  F  ~10대     1  41.40  0.37  1608.56    18
3  2021-06-01     0   2919    정기권  F  ~10대     1   0.00  0.00     0.00    75
4  2021-06-01     0    549    정기권  F  ~10대     1  13.04  0.17   731.55     6

## Question 31
### **대여일자별 데이터의 수를 데이터프레임으로 출력하고, 가장 많은 데이터가 있는 날짜를 출력하라**

In [222]:
result = df.groupby('대여일자').size()
print(result)
Ans = result.sort_values(ascending = False).index[0]
print(Ans)

대여일자
2021-06-01    48081
2021-06-02    56100
2021-06-03    18985
2021-06-04    56788
2021-06-05    52730
2021-06-06    47787
2021-06-07    52623
2021-06-08    55623
2021-06-09    54507
2021-06-10    36992
2021-06-11    52089
2021-06-12    48569
2021-06-13    43131
2021-06-14    54945
dtype: int64
2021-06-04


## Question 32
### **각 일자의 요일을 표기하고 ('Monday' ~'Sunday') 'day_name'컬럼을 추가하고 이를 이용하여 각 요일별 이용 횟수의 총합을 데이터 프레임으로 출력하라**

In [226]:
df['대여일자'] = df.대여일자.astype('datetime64')

C:\Users\USER-PC\AppData\Local\Temp\ipykernel_31640\2178666635.py:1: FutureWarning: Passing unit-less datetime64 dtype to .astype is deprecated and will raise in a future version. Pass 'datetime64[ns]' instead
  df['대여일자'] = df.대여일자.astype('datetime64')


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 678950 entries, 0 to 678949
Data columns (total 11 columns):
 #   Column  Non-Null Count   Dtype         
---  ------  --------------   -----         
 0   대여일자    678950 non-null  datetime64[ns]
 1   대여시간    678950 non-null  int64         
 2   대여소번호   678950 non-null  int64         
 3   대여구분코드  678950 non-null  object        
 4   성별      678950 non-null  object        
 5   연령대코드   678950 non-null  object        
 6   이용건수    678950 non-null  int64         
 7   운동량     678950 non-null  object        
 8   탄소량     678950 non-null  object        
 9   이동거리    678950 non-null  float64       
 10  사용시간    678950 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(4), object(5)
memory usage: 57.0+ MB


In [239]:
df['day_name'] = df.대여일자.dt.day_name()
Ans1 = df.groupby('day_name').size().sort_values(ascending = False)
Ans2 = df.day_name.value_counts().to_frame()

print(Ans1)
print(Ans2)

day_name
Wednesday    110607
Friday       108877
Monday       107568
Tuesday      103704
Saturday     101299
Sunday        90918
Thursday      55977
dtype: int64
           day_name
Wednesday    110607
Friday       108877
Monday       107568
Tuesday      103704
Saturday     101299
Sunday        90918
Thursday      55977


## Question 33
### **각 요일별 가장 많이 이용한 대여소의 이용횟수와 대여소 번호를 데이터 프레임으로 출력하라**

## Question 34
### **나이대별 대여구분 코드의 (일일권/전체횟수) 비율을 구한 후 가장 높은 비율을 가지는 나이대를 확인하라. 일일권의 경우 일일권 과 일일권(비회원)을 모두 포함하라**

## Question 35
### **연령대별 평균 이동거리를 구하여라**

## Question 36
### **연령대 코드가 20대인 데이터를 추출하고,이동거리값이 추출한 데이터의 이동거리값의 평균 이상인 데이터를 추출한다.최종 추출된 데이터를 대여일자, 대여소 번호 순서로 내림차순 정렬 후 1행부터 200행까지의 탄소량의 평균을 소숫점 3째 자리까지 구하여라**

## Question 37
### **6월 7일 ~10대의 "이용건수"의 중앙값은?**

## Question 38
### **평일 (월~금) 출근 시간대(오전 6,7,8시)의 대여소별 이용 횟수를 구해서 데이터 프레임 형태로 표현한 후 각 대여시간별 이용 횟수의 상위 3개 대여소와 이용횟수를 출력하라**

## Question 39
### **이동거리의 평균 이상의 이동거리 값을 가지는 데이터를 추출하여 추출데이터의 이동거리의 표본표준편차 값을 구하여라**

## Question 40
### **남성('M' or 'm')과 여성('F' or 'f')의 이동거리값의 평균값을 구하여라**

### 전세계 행복도 지표 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/unsdsn/world-happiness(참고, 데이터 수정)     
데이터 설명 : 전세계 행복도 지표 조사     
data url = https://raw.githubusercontent.com/Datamanim/datarepo/main/happy2/happiness.csv       
```

In [ ]:
import pandas as pd
df =pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/happy2/happiness.csv',encoding='utf-8')
df.head()

## Question 41
### **데이터는 2018년도와 2019년도의 전세계 행복 지수를 표현한다. 각년도의 행복랭킹 10위를 차지한 나라의 행복점수의 평균을 구하여라**

## Question 42
### **데이터는 2018년도와 2019년도의 전세계 행복 지수를 표현한다. 각년도의 행복랭킹 50위이내의 나라들의 각각의 행복점수 평균을 데이터프레임으로 표시하라**

## Question 43
### **2018년도 데이터들만 추출하여 행복점수와 부패에 대한 인식에 대한 상관계수를 구하여라**

## Question 44
### **2018년도와 2019년도의 행복랭킹이 변화하지 않은 나라명의 수를 구하여라**

## Question 45
### **2019년도 데이터들만 추출하여 각변수간 상관계수를 구하고 내림차순으로 정렬한 후 상위 5개를 데이터 프레임으로 출력하라. 컬럼명은 v1,v2,corr으로 표시하라**

## Question 46
### **각 년도별 하위 행복점수의 하위 5개 국가의 평균 행복점수를 구하여라**

## Question 47
### **2019년 데이터를 추출하고 해당데이터의 상대 GDP 평균 이상의 나라들과 평균 이하의 나라들의 행복점수 평균을 각각 구하고 그 차이값을 출력하라**

## Question 48
### **각년도의 부패에 대한인식을 내림차순 정렬했을때 상위 20개 국가의 부패에 대한인식의 평균을 구하여라**

## Question 49
### **2018년도 행복랭킹 50위 이내에 포함됐다가 2019년 50위 밖으로 밀려난 국가의 숫자를 구하여라**

## Question 50
### **2018년,2019년 모두 기록이 있는 나라들 중 년도별 행복점수가 가장 증가한 나라와 그 증가 수치는?**

### 지역구 에너지 소비량 데이터

```{attention} 
데이터 출처 :       https://archive.ics.uci.edu/ml/datasets/Power+consumption+of+Tetouan+city (참고, 데이터 수정)     
데이터 설명 :  기온, 습도,바람풍속에 따른 도시의 3개 지역구의 에너지 소비량     
data url = https://raw.githubusercontent.com/Datamanim/datarepo/main/consum/Tetuan%20City%20power%20consumption.csv     
```

In [ ]:
import pandas as pd
df= pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/consum/Tetuan%20City%20power%20consumption.csv')
df.head()

## Question 51
### **DateTime컬럼을 통해 각 월별로 몇개의 데이터가 있는지 데이터 프레임으로 구하여라**

## Question 52
### **3월달의 각 시간대별 온도의 평균들 중 가장 낮은 시간대의 온도를 출력하라**

## Question 53
### **3월달의 각 시간대별 온도의 평균들 중 가장 높은 시간대의 온도를 출력하라**

## Question 54
### **Zone 1 Power Consumption 컬럼의 value값의 크기가 Zone 2  Power Consumption 컬럼의 value값의 크기보다 큰 데이터들의 Humidity의 평균을 구하여라**

## Question 55
### **각 zone의 에너지 소비량의 상관관계를 구해서 데이터 프레임으로 표기하라**

## Question 56
### **Temperature의 값이 10미만의 경우 A, 10이상 20미만의 경우 B,20이상 30미만의 경우 C, 그 외의 경우 D라고 할때 각 단계의 데이터 숫자를 구하여라**

## Question 57
### **6월 데이터중 12시의 Temperature의 표준편차를 구하여라**

## Question 58
### **6월 데이터중 12시의 Temperature의 분산을 구하여라**

## Question 59
### **Temperature의 평균이상의 Temperature의 값을 가지는 데이터를 Temperature를 기준으로 정렬 했을때 4번째 행의 Humidity 값은?**

## Question 60
### Temperature의 중간값 이상의 Temperature의 값을 가지는 데이터를Temperature를 기준으로 정렬 했을때 4번째 행의 Humidity 값은?

### 포켓몬 정보 데이터

```{attention} 
데이터 출처 :   https://www.kaggle.com/abcsds/pokemon   (참고, 데이터 수정)    
데이터 설명 :  포켓몬 정보    
data url = https://raw.githubusercontent.com/Datamanim/datarepo/main/pok/Pokemon.csv 
```

In [ ]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/pok/Pokemon.csv')
df.head()

## Question 61
### **Legendary 컬럼은 전설포켓몬 유무를 나타낸다.전설포켓몬과 그렇지 않은 포켓몬들의 HP평균의 차이를 구하여라**

## Question 62
### **Type 1은 주속성 Type 2 는 부속성을 나타낸다. 가장 많은 부속성 종류는 무엇인가?**

## Question 63
### **가장 많은 Type 1 의 종의 평균 Attack 을 평균 Defense로 나눈값은?**

## Question 64
### **포켓몬 세대(Generation) 중 가장많은 Legendary를 보유한 세대는 몇세대인가?**

## Question 65
### **'HP', 'Attack', 'Defense', 'Sp. Atk', 'Sp. Def', 'Speed' 간의 상관 계수중 가장 절댓값이 큰 두 변수와 그 값을 구하여라**

## Question 66
### **각 Generation의 Attack으로 오름차순 정렬시 상위 3개 데이터들(18개)의 Attack의 전체 평균을 구하여라**

## Question 67
### **각 Generation의 Attack으로 내림차순 정렬시 상위 5개 데이터들(30개)의 Attack의 전체 평균을 구하여라**

## Question 68
### **가장 흔하게 발견되는 (Type1 , Type2) 의 쌍은 무엇인가?**

## Question 69
### **한번씩만 존재하는 (Type1 , Type2)의 쌍의 갯수는 몇개인가?**

## Question 70
### **한번씩만 존재하는 (Type1 , Type2)의 쌍을 각 세대(Generation)은 각각 몇개씩 가지고 있는가?**

### 대한민국 체력장 데이터

```{attention} 
데이터 출처 :   국민체육진흥공단 (문화 빅데이터플랫폼)   (참고, 데이터 수정)     
데이터 설명 : 대한민국 국민 체력장 평가     
data url = https://raw.githubusercontent.com/Datamanim/datarepo/main/body/body.csv     
```

In [ ]:
import pandas as pd
df =pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/body/body.csv')
df.head()

## Question 71
### **전체데이터의 수축기혈압(최고) - 이완기혈압(최저)의 평균을 구하여라**

## Question 72
### **50~59세의 신장평균을 구하여라**

## Question 73
### **연령대 (20~29 : 20대 ...) 별  인원수를 구하여라**

## Question 74
### **연령대 (20~29 : 20대 ...) 별 등급의 숫자를 데이터 프레임으로 표현하라**

## Question 75
### **남성 중 A등급과 D등급의 체지방률 평균의 차이(큰 값에서 작은 값의 차)를 구하여라**

## Question 76
### **여성 중 A등급과 D등급의 체중의 평균의 차이(큰 값에서 작은 값의 차)를 구하여라**

## Question 77
### **bmi는 자신의 몸무게(kg)를 키의 제곱(m)으로 나눈값이다. 데이터의 bmi 를 구한 새로운 컬럼을 만들고 남성의 bmi 평균을 구하여라**

## Question 78
### **bmi보다 체지방율이 높은 사람들의 체중평균을 구하여라**

## Question 79
### **남성과 여성의 악력 평균의 차이를 구하여라**

## Question 80
### **남성과 여성의 교차윗몸일으키기 횟수의 평균의 차이를 구하여라**

### 기온 강수량 데이터

```{attention} 
데이터 출처 : https://data.kma.go.kr/cmmn/static/staticPage.do?page=intro    
데이터 설명 : 2020년도 이화동(서울) , 수영동(부산)의 시간단위의 기온과 강수량    
데이터 url : https://raw.githubusercontent.com/Datamanim/datarepo/main/weather/weather2.csv    
```

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/weather/weather2.csv")
df.head()

## Question 81
### **여름철(6월,7월,8월) 이화동이 수영동보다 높은 기온을 가진 시간대는 몇개인가?**

## Question 82
### **이화동과 수영동의 최대강수량의 시간대를 각각 구하여라**

### 서비스 이탈예측 데이터

```{attention} 
데이터 출처 : https://www.kaggle.com/shubh0799/churn-modelling 에서 변형    
데이터 설명 :  고객의 신상정보 데이터를 통한 회사 서비스 이탈 예측 (종속변수 : Exited)     
Data url : https://raw.githubusercontent.com/Datamanim/datarepo/main/churn/train.csv     
```  

In [ ]:
import pandas as pd
#데이터 로드
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/churn/train.csv")
df.head()

## Question 83
### **남성 이탈(Exited)이 가장 많은 국가(Geography)는 어디이고 이탈 인원은 몇명인가?**

## Question 84
### **카드를 소유(HasCrCard ==1)하고 있으면서 활성멤버(IsActiveMember ==1) 인 고객들의 평균 나이를 소숫점이하 4자리까지 구하여라? **

## Question 85
### **Balance 값이 중간값 이상을 가지는 고객들의 CreditScore의 표준편차를 소숫점이하 3자리까지 구하여라**

### 성인 건강검진 데이터

```{attention} 
데이터 출처 :https://www.data.go.kr/data/15007122/fileData.do (참고, 데이터 수정)    
데이터 설명 : 2018년도 성인의 건강검 진데이터 (흡연상태 1- 흡연, 0-비흡연 )    
데이터 url : https://raw.githubusercontent.com/Datamanim/datarepo/main/smoke/train.csv     
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/smoke/train.csv")
df.head()

## Question 86
### **수축기혈압과 이완기 혈압기 수치의 차이를 새로운 컬럼('혈압차') 으로 생성하고, 연령대 코드별 각 그룹 중 '혈압차' 의 분산이 5번째로 큰 연령대 코드를 구하여라**

## Question 87
### **비만도를 나타내는 지표인 WHtR는 허리둘레 / 키로 표현한다. 일반적으로 0.58이상이면 비만으로 분류한다. 데이터중 WHtR 지표상 비만인 인원의 남/여 비율을 구하여라**

### 자동차 보험가입 예측데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/anmolkumar/health-insurance-cross-sell-prediction(참고, 데이터 수정)    
데이터 설명 : 자동차 보험 가입 예측    
Data url : https://raw.githubusercontent.com/Datamanim/datarepo/main/insurance/train.csv       
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/insurance/train.csv")
df.head()

## Question 88
### **Vehicle_Age 값이 2년 이상인 사람들만 필터링 하고 그중에서  Annual_Premium 값이 전체 데이터의 중간값 이상인 사람들을 찾고, 그들의 Vintage값의 평균을 구하여라**

## Question 89
### **vehicle_age에 따른 각 성별(gender)그룹의 Annual_Premium값의 평균을 구하여 아래 테이블과 동일하게 구현하라**

### 핸드폰 가격 예측데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/iabhishekofficial/mobile-price-classification?select=train.csv(참고, 데이터 수정)    
데이터 설명 : 핸드폰 가격예측 (price_range컬럼 0(저렴) ~3(매우비쌈) 범위 )         
Data url https://raw.githubusercontent.com/Datamanim/datarepo/main/mobile/train.csv      
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/mobile/train.csv")
df.head()

## Question 90
### **price_range 의 각 value를 그룹핑하여 각 그룹의 n_cores 의 빈도가 가장높은 value와 그 빈도수를 구하여라**

## Question 91
### **price_range 값이 3인 그룹에서 상관관계가 2번째로 높은 두 컬럼과 그 상관계수를 구하여라**

### 비행탑승 경험 만족도 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/teejmahal20/airline-passenger-satisfaction?select=train.csv (참고, 데이터 수정)     
데이터 설명 : 비행탑승 경험 만족도 (satisfaction 컬럼 : 'neutral or dissatisfied' or satisfied )   (83123, 24) shape     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/airline/train.csv       
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/airline/train.csv")
df.head()

## Question 92
### **Arrival Delay in Minutes 컬럼이 결측치인 데이터들 중 'neutral or dissatisfied' 보다 'satisfied'의 수가 더 높은 Class는 어디 인가?**

### 수질 음용성 여부 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/adityakadiwal/water-potability    
데이터 설명 : 수질 음용성 여부 (Potablillity 컬럼 : 0 ,1 )     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/waters/train.csv     
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/waters/train.csv")
df.head()

## Question 93
### **ph값은 상당히 많은 결측치를 포함한다. 결측치를 제외한 나머지 데이터들 중 사분위값 기준 하위 25%의 값들의 평균값은?**

### 의료 비용 예측 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/mirichoi0218/insurance/code(참고, 데이터 수정)    
데이터 설명 : 의료비용 예측문제     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/MedicalCost/train.csv     
```  

In [ ]:
import pandas as pd
train = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/MedicalCost/train.csv")
train.head()

## Question 94
### **흡연자와 비흡연자 각각 charges의 상위 10% 그룹의 평균의 차이는?**

### 킹카운티 주거지 가격예측문제 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/harlfoxem/housesalesprediction (참고, 데이터 수정)    
데이터 설명 : 킹카운티 주거지 가격 예측문제     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/kingcountyprice/train.csv         
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/kingcountyprice//train.csv")
df.head()

## Question 95
### **bedrooms 의 빈도가 가장 높은 값을 가지는 데이터들의 price의 상위 10%와 하위 10%값의 차이를 구하여라**

### 대학원 입학가능성 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/mohansacharya/graduate-admissions(참고, 데이터 수정)     
데이터 설명 : 대학원 입학 가능성 예측     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/admission/train.csv         
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/admission/train.csv")
df.head()

## Question 96
### **Serial No. 컬럼을 제외하고 'Chance of Admit'을 종속변수, 나머지 변수를 독립변수라 할때, 랜덤포레스트를 통해 회귀 예측을 할 떄 변수중요도 값을 출력하라 (시드값에 따라 순서는 달라질수 있음)**

### 레드 와인 퀄리티 예측 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/uciml/red-wine-quality-cortez-et-al-2009(참고, 데이터 수정)     
데이터 설명 : 레드 와인 퀄리티 예측문제     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/redwine/train.csv        
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/redwine/train.csv")
df.head()

## Question 97
### **quality 값이 3인 그룹과  8인 데이터그룹의 각 컬럼별 독립변수의 표준편차 값의 차이를 구할때 그값이 가장 큰 컬럼명을 구하여라**


### 약물 분류 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/prathamtripathi/drug-classification(참고, 데이터 수정)    
데이터 설명 : 투약하는 약을 분류     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/drug/train.csv         
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/drug/train.csv")
df.head()

## Question 98
### **남성들의 연령대별 (10살씩 구분 0~9세 10~19세 ...) Na_to_K값의 평균값을 구해서 데이터 프레임으로 표현하여라**

### 사기회사 분류 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/sid321axn/audit-data(참고, 데이터 수정)    
데이터 설명 : 사기회사 분류    
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/audit/train.csv          
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/audit/train.csv")
df.head()

## Question 99
### **데이터의 Risk 값에 따른 score_a와 score_b의 평균값을 구하여라**

### 센서데이터 동작유형 분류 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/kyr7plus/emg-4(참고, 데이터 수정)    
데이터 설명 : 센서데이터로 동작 유형 분류     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/muscle/train.csv       
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/muscle/train.csv")
df.head()

## Question 100
### **pose값에 따른 각 motion컬럼의 중간값의 가장 큰 차이를 보이는 motion컬럼은 어디이며 그값은?**

### 현대 차량 가격 분류문제 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/mysarahmadbhat/hyundai-used-car-listing(참고, 데이터 수정)     
데이터 설명 : 현대 차량가격 분류문제     
Data url :hhttps://raw.githubusercontent.com/Datamanim/datarepo/main/hyundai/train.csv        
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/hyundai/train.csv")
df.head()

## Question 101
### **정보(row수)가 가장 많은 상위 3차종의 price값의 각 평균값은?**

### 당뇨여부판단 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/pritsheta/diabetes-dataset(참고, 데이터 수정)     
데이터 설명 :  당뇨여부 판단하기     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/diabetes/train.csv      
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/diabetes/train.csv")
df.head()

## Question 102
### **Outcome 값에 따른 각 그룹의  각 컬럼의 평균 차이를 구하여라**

### 넷플릭스 주식 데이터 

```{attention} 
데이터 출처 :https://www.kaggle.com/pritsheta/netflix-stock-data-from-2002-to-2021(참고, 데이터 수정)     
데이터 설명 : 넷플릭스 주식데이터     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/nflx/NFLX.csv         
```  

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/nflx/NFLX.csv")
df.head()

## Question 103
### **매년 5월달의 open가격의 평균값을 데이터 프레임으로 표현하라**

## 220510추가

### nba 선수 능력치 데이터

```{attention} 
데이터 출처 :https://www.kaggle.com/datasets/vivovinco/nba-player-stats(참고, 데이터 수정)     
Data url :https://raw.githubusercontent.com/Datamanim/datarepo/main/nba/nba.csv         
```  

In [ ]:
pd.set_option('display.max_columns',50)
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/nba/nba.csv",encoding='latin',sep=';')
df.head()

## Question 104
### **Tm 컬럼은 각 팀의 이름을 의미한다. TOR팀의 평균나이를 소수 4째 자리까지 구하여라**

## Question 105
### **Pos 컬럼은 포지션을 의미한다. 전체 선수 중 최소나이대의 선수들을 필터하고 그들 중 가장 많은 포지션은 무엇인지 확인하라**

## Question 106
### **선수들의 이름은 first_name+ 공백 + last_name으로 이루어져 있다. 가장 많은 first_name은 무엇이며 몇 회 발생하는지 확인하라**

## Question 107
### **PTS컬럼은 경기당 평균득점수 이다. 각포지션별로 경기당 평균득점수의 평균을 구하여라**

## Question 108
### **PTS컬럼은 경기당 평균득점수 이다. 각포지션별로 경기당 평균득점수의 평균을 구하여라**

## Question 109
### **G컬럼은 참여한 경기의 숫자이다. 각 팀별로 가장 높은 경기참여 수를 가진 선수들의 경기 참여 숫자의 평균을 구하여라**

## Question 110
### **Tm의 값이 MIA이며 Pos는 C또는 PF인 선수의 MP값의 평균은?**

## Question 111
### **전체 데이터중 G의 평균값의 1.5배 이상인 데이터들만 추출했을때 3P값의 평균은?**

## Question 112
### **Age의 평균 이상인 그룹과 평균 미만인 그룹간의 G값의 평균의 차이는?**

## Question 113
### **평균나이가 가장 젊은 팀은 어디인가**

## Question 114
### **Pos그룹별 평균 MP값을 구하여라**